### PR TITLE
rewrite code to escape await logic

### PR DESCRIPTION
**Pull request by NailxSharipov**
write now I have a very weird situation:
A lot of awaiting threads
[screen](https://ibb.co/QMYDzZX)

### CODE REVIEW

File: Sources/SwiftCentrifuge/Client.swift [hunk 1]
```diff
@@ -450,121 +450,53 @@ fileprivate extension CentrifugeClient {
...
-                }
-                if timedOut {
-                    completion(nil, CentrifugeError.timeout)
-                    return
-                }
-                completion(reply, nil)
+        self.opCallbacks[id] = { [weak self] rep in
+            timeoutTask.cancel()
+            
+            self?.opCallbacks[id] = nil
```
Comment (NailxSharipov) [inline Sources/SwiftCentrifuge/Client.swift]: I just understood that remove from dictionary is better something into the 'handleData' function

**Comment by NailxSharipov** [comment]
I rewrote all logic with timeout task. All that code are working on sync queue, so you don't need any await in that logic. Which closure had run the first would cancel the other one.

File: Sources/SwiftCentrifuge/Client.swift (hunk 1 above)
Comment (NailxSharipov) [reply Sources/SwiftCentrifuge/Client.swift]: After deep researching I understood that in this case we must remove callback from dictionary by the hands. But in others same cases It's unnecessary.

**Comment by FZambia** [comment]
Looks very good, thanks! Not sure why there was a code with `DispatchGroup` - most probably it was required on early development stages where not all places used completion callbacks. But I can't say for sure at moment.

**Pull request merged**